build.gradle

```diff
@@ -41,6 +52,110 @@ subprojects {
+
+        task copyScripts(type: Copy){
+            mkdir "$projectDir/scripts/"
+            from "${projectDir}/../../k8s-deployment/common/"
+            into "$projectDir/scripts/"
+        }
+
+        task dockerfile(type: Dockerfile) {
+            dependsOn copyScripts
+            dependsOn fatJar
+            destFile = project.file('./Dockerfile')
+            from 'alpine:3.7'
+            label(['maintainer': 'huawei@huawei.com'])
+            runCommand  'apk --update add openjdk8-jre bind-tools'
+            // add this statement for respective Application
+            //copyFile './build/libs/${ProjectName}.jar', '/root/amino/jars/'
+            copyFile './scripts/*', '/root/amino/'
+        }
+
+        task buildDockerImage(type: DockerBuildImage) {
+            dependsOn dockerfile
+            url = 'unix:///var/run/docker.sock'
+            inputDir = file("$projectDir")
+        }
+
+        task pushImage(type: DockerPushImage) {
+            dependsOn buildDockerImage
+        }
+
+        task createOMSDeployment (type: Copy) {
+            from "${projectDir}/../../k8s-deployment/"
+            into "$projectDir"
+            include 'oms.template.yml'
+            rename { file -> 'oms.yml' }
+        }
+
+        task createKernelServerDeployment (type: Copy) {
+            from "${projectDir}/../../k8s-deployment/"
+            into "$projectDir"
+            include 'kernelserver.template.yml'
+            rename { file -> 'kernelserver.yml' }
+        }
+
+        deployer {
+            kubernetes {
+                id = project.property('omsId')
+                baseUrl = project.property('baseUrl')
+                namespace = project.property('namespace')
+                strategy = project.property('strategy')
+                auth {
+                    username = project.property('k8s.username')
+                    password = project.property('k8s.password')
+                }
+                ssl {
+                    trustAll = true
+                }
+                files = [file("oms.yml")]
+            }
+        }
+
+        deployer {
+            kubernetes {
+                id = project.property('kernelServerId')
+                baseUrl = project.property('baseUrl')
+                namespace = project.property('namespace')
+                strategy = project.property('strategy')
+                auth {
+                    username = project.property('k8s.username')
+                    password = project.property('k8s.password')
+                }
+                ssl {
+                    trustAll = true
+                }
+                files = [file("kernelserver.yml")]
+            }
+        }
+
+        task deployOms(type: Exec){
+            dependsOn pushImage
+            dependsOn createOMSDeployment
+            commandLine "${projectDir}/../../gradlew", 'deploy', "--environmentId=oms"
+        }
+
+        task deployKernelServer(type: Exec){
+            dependsOn deployOms
```

Comment:
I think, deploy OMS and deploy kernel server(1 at a time) being separate is better instead of this one task deploying OMS and 2 kernel servers implicitly. User can deploy  OMS first and then deploy kernel server( any number of times) depending on the need ? Incorrectly deploying kernel server without deploying OMS can be detected in that case ? You may consider it in next PR if the current PR size is going to increase a lot.

Reply:
Change made as per suggestion, can be seen in commit 0a251f3 PR #101